static/app/views/organizationStats/usageChart/utils.tsx

```diff
@@ -83,58 +83,52 @@ export function getTooltipFormatter(dataCategory: DataCategoryInfo['plural']) {
 const MAX_NUMBER_OF_LABELS = 10;
 
 /**
+ * Determines which X-axis labels should be visible based on data period and intervals.
  *
  * @param dataPeriod - Quantity of hours covered by the data
- * @param numBars - Quantity of data points covered by the dataPeriod
+ * @param intervals - Intervals to be displayed on the X-axis
+ * @returns An object containing an array indicating visibility of each X-axis label
  */
-export function getXAxisLabelInterval(dataPeriod: number, numBars: number) {
-  return dataPeriod > 7 * 24
-    ? getLabelIntervalLongPeriod(dataPeriod, numBars)
-    : getLabelIntervalShortPeriod(dataPeriod, numBars);
-}
-
-/**
- * @param dataPeriod - Quantity of hours covered by data, expected 7+ days
- */
-function getLabelIntervalLongPeriod(dataPeriod: number, numBars: number) {
-  const days = dataPeriod / 24;
-  if (days <= 7) {
-    throw new Error('This method should be used for periods > 7 days');
+export function getXAxisLabelVisibility(dataPeriod: number, intervals: string[]) {
+  if (dataPeriod <= 24) {
+    return {
+      xAxisLabelVisibility: Array(intervals.length).fill(false),
+    };
   }
 
-  // Use 1 tick per day
-  let numTicks = days;
-  let numLabels = numTicks;
-
-  const daysBetweenLabels = [2, 4, 7, 14];
-  const daysBetweenTicks = [1, 2, 7, 7];
-
-  for (let i = 0; i < daysBetweenLabels.length && numLabels > MAX_NUMBER_OF_LABELS; i++) {
-    numLabels = numTicks / daysBetweenLabels[i];
-    numTicks = days / daysBetweenTicks[i];
+  const uniqueLabels: string[] = [];
+  const labelToPositionMap: Record<number, boolean> = {};
+  const labelVisibility: boolean[] = new Array(intervals.length).fill(false);
+
+  // Collect unique labels and their positions
+  intervals.forEach((label, index) => {
+    if (index === 0 || label.slice(0, 6) !== intervals[index - 1].slice(0, 6)) {
```

Comment:
slice returns a new array and will never return true
`label.slice(0, 6) !== intervals[index - 1].slice(0, 6)`

Reply:
I understand your concern, but in this case we are comparing 2x strings. When comparing strings, JavaScript compares their content directly, not their references. If two strings have the same content, they are considered equal. Does that make sense?

Reply:
the result of the code below for example is `false`

```
const label = 'Aug 6'
const secondLabel = 'Aug 6'

console.log(label.slice(0, 6) !== secondLabel.slice(0, 6))
```

Reply:
Oh it's a string lol